framework/src/meshgenerators/CircularBoundaryCorrectionGenerator.C

```diff
@@ -0,0 +1,426 @@
+      _input_mesh_circular_bids.size());
+  // Loop over all the boundary sides and store the sides (node pairs) on each of the circular
+  // boundaries to be corrected
+  for (unsigned int i = 0; i < side_list.size(); ++i)
+  {
+    auto side_list_it = std::find(_input_mesh_circular_bids.begin(),
+                                  _input_mesh_circular_bids.end(),
+                                  std::get<2>(side_list[i]));
+    if (side_list_it != _input_mesh_circular_bids.end())
+    {
+      const auto side =
+          input_mesh->elem_ptr(std::get<0>(side_list[i]))->side_ptr(std::get<1>(side_list[i]));
+      input_circ_bds_sds[std::distance(_input_mesh_circular_bids.begin(), side_list_it)].push_back(
+          std::make_pair(side->point(0), side->point(1)));
+      input_circ_bds_sds_ids[std::distance(_input_mesh_circular_bids.begin(), side_list_it)]
+          .push_back(std::make_pair(side->node_id(0), side->node_id(1)));
+    }
+  }
+
+  // Computes the correction coefficient and apply it;
+  // Also records the nodes that have been modified to avert double-correction
+  std::set<dof_id_type> mod_node_list;
+  for (unsigned int i = 0; i < input_circ_bds_pts.size(); i++)
+  {
+    auto & input_circ_bd_pts = input_circ_bds_pts[i];
+    if (input_circ_bd_pts.size() < 3)
+      paramError("input_mesh_circular_boundaries",
+                 "too few nodes in one of the provided boundaries.");
+    Real bdry_rad;
+    const Point boundary_origin =
+        circularCenterCalculator(input_circ_bd_pts, bdry_rad, _custom_circular_tolerance[i]);
+
+    // Check if the boundary is a full circle or partial
+    // Also make an ordered array of nodes
+    Real dummy_max_rad;
+    std::vector<dof_id_type> ordered_node_list;
+    bool is_bdry_closed;
+    try
+    {
+      FillBetweenPointVectorsTools::isClosedLoop(*input_mesh,
+                                                 dummy_max_rad,
+                                                 ordered_node_list,
+                                                 input_circ_bds_sds_ids[i],
+                                                 boundary_origin,
+                                                 "boundary",
+                                                 is_bdry_closed);
+    }
+    catch (const MooseException & e)
+    {
+    }
+
+    // If the user selects to move the end nodes of a partial circular boundary, we need to
+    // calculate the displacement of the end nodes, which differs from the displacement of the
+    // other nodes
+    Real end_node_disp;
+    // c_coeff is the coefficient used to scale the azimuthal angles
+    // corr_factor > 1 means the partial boundary is less than a half circle
+    // corr_factor < 1 means the partial boundary is more than a half circle
+    // corr_factor = 1 means the boundary is a full circle or a half circle
+    Real c_coeff;
+    const Real corr_factor = generalCirCorrFactor(
```

Comment:
why does this not depend on whether the end nodes are moved?
isnt the area going to depend on where the end nodes are?

Reply:
missed that in the `if`. Will fix this.